datajoint/heading.py

```diff
@@ -120,7 +121,7 @@ def table_status(self):
     def attributes(self):
         if self._attributes is None:
             self._init_from_database()  # lazy loading from database
-        return self._attributes
+        return {k: v for k, v in self._attributes.items() if not v.is_hidden}
```

Comment:
Note: this changes from direct access to the `_attributes` dictionary to a copy. This means that the following will raise AssertionError as of this PR, but not on `master`:

```python
orig_value = my_table.heading.attributes['my_attr']
my_table.heading.attributes['my_attr'] = "foo"
assert my_table.heading.attributes['my_attr'] == "foo"
assert my_table.heading.attributes['my_attr'] != orig_value
```